base/exports.jl

```diff
@@ -993,6 +993,7 @@ export
     error,
     rethrow,
     retry,
+    default_retry_check,
```

Comment:
@amitmurthy i'd be happy not exporting `default_retry_check`.  what do you think?